tests/data/wire/ext_conf_invalid_requested_version.xml

```diff
@@ -0,0 +1,38 @@
+<Extensions version="1.0.0.0" goalStateIncarnation="9">
```

Comment:
since vmsettings will be our primary source for extensions goal state we should start using those for our test data

Reply:
I need to do the global change soon, i am already started having problems with this in my latest PR. I'll do the global change as part of Fast Track, but it'd help if we start doing this for new tests, like this one.

Reply:
Sure sure, happy to do that. Can you share some guidance for me though, like what exactly do i need to do here? I want my changes to be in sync with the global changes that come in later.

Reply:
you know what? you got it me here because I just realized that no matter what you do most likely it's going to break with my next change so I guess I'll do this change as well then. good as it is now

just fyi: right now all the test that use the mock protocol hit update_goal_state and use mock data that relies on extensionsConfig. Since we do not have a matching vmSettings for those mock data, we hit a goal state mistmach and we fallback to extensionsConfig, which what those tests need anyway.

this is going to break in my next iteration, when there will be no fallback to extensionsConfig